### ISSUE TITLE
Documentation enhancement - document the returned data

### ISSUE THREAD

**Issue by ThoreKr** [Lhdang88/cloud_computing_ws#3]
To enhance the documentation it would be really helpful to also have some information about what kind of data the backend returns on the different API requests.

**Comment by Sincap80** [comment]
Web socket communication: String: 
message: { 
        data: "xyz"
}
message.data contains the value returned from the Server.

Return values from server:
getMessagesInRoom: message
postMessage: sent message
postPrivateMessage:  sent message
getUsersInRoom: list of users
getChatRooms: list of rooms

A console.log of the received data would help.

**Comment by ThoreKr** [comment]
@Sincap80 Thanks for repeating the readme.
Of course one could use curl and manually find out what the different API calls do - but reverse engineering the backend shouldn't be the job of the frontend developers (unless you want this to happen: http://www.commitstrip.com/en/2017/02/20/no-ones-fault/).

**Comment by Lhdang88** [comment]
you can contribute and create a pull request ...

**Comment by ThoreKr** [comment]
I'm sorry, but this is the second really poor response.
The assignment was to develop a front-end for a given backend.
The course (as we were told several times) was clearly defined to not cover the server-side.

Writing an own API documentation is clearly out of the given scope of the assignment.

**Comment by ThoreKr** [comment]
If developed properly, using tools such as [swagger](https://swagger.io/) or [apiDoc](http://apidocjs.com/) most of  the documentation would have happened automagically.

**Comment by Lhdang88** [comment]
@ThoreKr 
This is a very idealistic view. Unfortunately, in open-source software, it is very common that documentation is lacking unless you have contributors who are willing to commit. This is an open-source project, where I do this stuff in my own freetime. Unless you are willing to contribute, you may have to wait for it. I have no commitment to support your university course.

**Comment by ThoreKr** [comment]
@Lhdang88 Then I have to apologize to make you responsible for it.
If it is your freetime project then @1357111317 is to blame for giving wrong information and defining a poor scope for this project.

**Comment by M320Trololol** [comment]
I agree with @ThoreKr here. We were told (by our University course lecturer) that we would be provided a working and documented backend for our chatclient (our course explicitly doesn't cover server side stuff) within two weeks of the start of the lecture (8th of May) that would be developed by one of his colleagues at Bosch. Since we have to submit our client on 17th of July, we kind of are in a hurry here.
